s4/clarity/lims.py

```diff
@@ -91,61 +92,102 @@ def __init__(self, root_uri, username, password, dry_run=False, insecure=False,
 
-        self.researchers = ElementFactory(self, Researcher, batch_flags=BatchFlags.QUERY)
+        self.researchers = ElementFactory(self, Researcher)
 
-        self.labs = ElementFactory(self, Lab, batch_flags=BatchFlags.QUERY)
+        self.labs = ElementFactory(self, Lab)
 
-        self.roles = ElementFactory(self, Role, batch_flags=BatchFlags.QUERY)
+        self.roles = ElementFactory(self, Role)
 
-        self.permissions = ElementFactory(self, Permission, batch_flags=BatchFlags.QUERY)
+        self.permissions = ElementFactory(self, Permission)
 
         # configuration
         from .configuration import Workflow, Protocol, ProcessType, Udf, ProcessTemplate, Automation
 
-        self.workflows = ElementFactory(self, Workflow, batch_flags=BatchFlags.QUERY,
-                                        request_path='/configuration/workflows')
-        self.protocols = ElementFactory(self, Protocol, batch_flags=BatchFlags.QUERY,
-                                        request_path='/configuration/protocols')
-        self.udfs = UdfFactory(self, Udf, batch_flags=BatchFlags.QUERY,
-                               request_path='/configuration/udfs')
-        self.process_types = ElementFactory(self, ProcessType, batch_flags=BatchFlags.QUERY,
-                                            name_attribute="displayname")
-        self.process_templates = ElementFactory(self, ProcessTemplate, batch_flags=BatchFlags.QUERY,
-                                                name_attribute="name")
-        self.automations = ElementFactory(self, Automation, batch_flags=BatchFlags.QUERY,
-                                          name_attribute="name", request_path="/configuration/automations")
+        self.workflows = ElementFactory(self, Workflow)
+
+        self.protocols = ElementFactory(self, Protocol)
+
+        self.udfs = UdfFactory(self, Udf)
+
+        self.process_types = ElementFactory(self, ProcessType)
+
+        self.process_templates = ElementFactory(self, ProcessTemplate)
+
+        self.automations = ElementFactory(self, Automation)
 
         self.stages = ElementFactory(self, Stage)
 
+    def _get_hostname(self):
+        # type: () -> str
+        """
+        Verify that we have a valid hostname in our root uri
+        and isolate the host name from it.
+        :return: The name of the Clarity server.
+        """
+
+        # Compile a regex statement to isolate the host name
+        # and verify that our root uri looks like a properly formed URI
+        hostname_finding_regex = re.compile(r'https?://([^/:]+)')
+
+        # Then use it to validate our uri
+        hostname_match = hostname_finding_regex.match(self.root_uri)
+        if not hostname_match:
+            raise Exception("No hostname found in LIMS uri: %s" % self.root_uri)
+
+        # Using the same RegEx query select out the hostname from the URI
+        return hostname_match.group(1)
+
+    def _get_environment(self):
```

Comment:
Now that this is a function and nice and clean, it seems like this should be overrideable by passing a param into the constructor, so I can tell the LIMS object it is a test LIMS explicitly.

Reply:
Yeah.... I mean the whole thing is pretty suspect in my opinion. I don't think that we can actually declare the machine objective based on the name. Like, if we get a client called GeneTest and they get a server called clarity.genetest.com we will think that every machine, other than their dev machine, is the test machine. 
Do we ever consume the output of this variable? It is unrelated to communicating with Clarity.
I would vote to remove it.

Reply:
You're probably right. Personally I wouldn't bother trying to remove it as part of this PR since this has been in limbo for a while and should get merged, might as well do it separately so we can argue about it there instead. (or I can after this is merged)